From 6f9017dc4c39f0d1f7d8cee431c10760539f46ea Mon Sep 17 00:00:00 2001
From: Orgad Shaneh <orgad.shaneh@audiocodes.com>
Date: Mon, 6 Jun 2022 15:33:40 +0300
Subject: [PATCH] Fix incomplete type error when using list with pair

Example:

boost::container::list<std::pair<int, int>> list;

.../boost/container/allocator_traits.hpp:403:11: error: invalid use of incomplete type 'struct boost::container::dtl::pair<int, int>'
  403 |    {  p->~T(); (void)p;  }
      |       ~~~~^
In file included from .../boost/container/detail/node_alloc_holder.hpp:39,
                 from .../boost/container/list.hpp:34:
.../boost/container/detail/is_pair.hpp:59:8: note: declaration of 'struct boost::container::dtl::pair<int, int>'
   59 | struct pair;
      |        ^~~~
---
 boost/container/list.hpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/boost/container/list.hpp b/boost/container/list.hpp
index 88fa4c3..23015ec 100644
--- a/boost/container/list.hpp
+++ b/boost/container/list.hpp
@@ -32,6 +32,7 @@
 #include <boost/container/detail/iterators.hpp>
 #include <boost/container/detail/mpl.hpp>
 #include <boost/container/detail/node_alloc_holder.hpp>
+#include <boost/container/detail/pair.hpp>
 #include <boost/container/detail/version_type.hpp>
 #include <boost/container/detail/value_functors.hpp>
 // move
-- 
2.36.1.windows.1

